src/main/java/com/minecolonies/coremod/client/gui/modules/PlantationFieldsModuleWindow.java

```diff
@@ -0,0 +1,232 @@
+
+        fieldList = findPaneOfTypeByID(LIST_FIELDS, ScrollingList.class);
+        fieldList.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return moduleView.getFields().size();
+            }
+
+            @Override
+            public void updateElement(final int index, @NotNull final Pane rowPane)
+            {
+                final PlantationField.View field = ((PlantationField.View) moduleView.getFields().get(index));
+                final String distance = Integer.toString(field.getDistance(buildingView));
+                final Component direction = BlockPosUtil.calcDirection(buildingView.getPosition(), field.getPosition());
+
+                final boolean canAddField = moduleView.canAssignField(field);
+
+                rowPane.findPaneOfTypeByID(TAG_DISTANCE, Text.class).setText(Component.translatable(distance + "m"));
+                rowPane.findPaneOfTypeByID(TAG_DIRECTION, Text.class).setText(direction);
+
+                final Button assignButton = rowPane.findPaneOfTypeByID(TAG_BUTTON_ASSIGN, Button.class);
+                assignButton.setEnabled(moduleView.assignFieldManually());
+                assignButton.setHoverPane(null);
+                assignButton.show();
+
+                if (field.isTaken() && !buildingView.getID().equals(field.getBuildingId()))
+                {
+                    assignButton.hide();
+                }
+                else
+                {
+                    if (field.isTaken())
+                    {
+                        assignButton.setText(Component.translatable(RED_X).withStyle(ChatFormatting.RED));
+                    }
+                    else
+                    {
+                        assignButton.setText(Component.translatable(APPROVE).withStyle(ChatFormatting.GREEN));
+
+                        if (!canAddField)
+                        {
+                            assignButton.disable();
+
+                            MutableComponent warningTooltip = moduleView.getFieldWarningTooltip(field);
+                            if (warningTooltip != null && moduleView.assignFieldManually())
+                            {
+                                PaneBuilders.tooltipBuilder()
+                                  .append(warningTooltip.withStyle(ChatFormatting.RED))
+                                  .hoverPane(assignButton)
+                                  .build();
+                            }
+                        }
+                    }
+                }
+
+                final Item item = PlantationModuleRegistry.getPlantationModule(field.getPlantationFieldType()).getItem();
+                rowPane.findPaneOfTypeByID(TAG_ICON, ItemIcon.class).setItem(new ItemStack(item));
+            }
+        });
+
+        updateUI();
+    }
+
+    @Override
+    public void onUpdate()
+    {
+        super.onUpdate();
+        updateUI();
```

Comment:
Whats the dif of the farmfieldsmodule and plantationfieldsmodule windows?

Reply:
Slightly different UI conditions, for example the farmer no longer has a plant limit, that has moved solely to the plantation.
The changes are marginal, but instead of making it so there's a few methods which change functionality, I figured completely extracting the classes separately would give more freedom (also later for pastures)